src/main/java/org/opentosca/csarrepo/service/DeployToOpenToscaService.java

```diff
@@ -0,0 +1,60 @@
+package org.opentosca.csarrepo.service;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opentosca.csarrepo.exception.DeploymentException;
+import org.opentosca.csarrepo.exception.PersistenceException;
+import org.opentosca.csarrepo.model.CsarFile;
+import org.opentosca.csarrepo.model.OpenToscaServer;
+import org.opentosca.csarrepo.model.repository.CsarFileRepository;
+import org.opentosca.csarrepo.model.repository.OpenToscaServerRepository;
+import org.opentosca.csarrepo.util.ContainerApiClient;
+
+public class DeployToOpenToscaService extends AbstractService {
+
+	private boolean success = false;
+	private static final Logger LOGGER = LogManager.getLogger(DeployToOpenToscaService.class);
+
+	public DeployToOpenToscaService(long userId, long openToscaId, long csarFileId) {
+		super(userId);
+
+		OpenToscaServerRepository openToscaServerRepository = new OpenToscaServerRepository();
+		CsarFileRepository csarFileRepo = new CsarFileRepository();
+
+		OpenToscaServer openToscaServer;
+		CsarFile csarFile;
+		try {
+			openToscaServer = openToscaServerRepository.getbyId(openToscaId);
+			csarFile = csarFileRepo.getbyId(csarFileId);
+		} catch (PersistenceException e) {
+			this.addError("Couldn't determine required parameters " + e.getMessage());
+			return;
+		}
+
+		try {
+			URL address = openToscaServer.getAddress();
+			// TODO: check if it would be better to save address as URI in
+			// general
+			ContainerApiClient containerApiClient = new ContainerApiClient(address.toURI());
+			String location = containerApiClient.uploadCSAR(csarFile);
+
+			// create meta-data
+			csarFile.addOpenToscaServer(openToscaServer, location);
+			csarFileRepo.save(csarFile);
```

Comment:
Is this really enough? do we need to add it from the other side to?
Regarding the creation in the database this doesn't matter. I'm just not sure what would happen if I follow the link from the OpenToscaServer-side? Would it be set or would it only be set after a reboot :question: